### PR TITLE
[DOC-295] [DOC-432]JumpCloud IdP for YBA

### PR DESCRIPTION
**Pull request by aishwarya24**
Deploy preview - https://deploy-preview-23214--infallible-bardeen-164bc9.netlify.app/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud/

### CODE REVIEW

**Review by Arpit-yb — CHANGES_REQUESTED** [review]
we are putting the title as - configuring OIDC in YBA on universe
However throughout the doc we are trying to setup the YBA (not universe) side authentication.
So a bit confused on what we want to cover as part of this doc change.
We have two types of oidc auths -
YBA side
Universe side
Both have different setup steps from our side (from jumcpcloud side its same)

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+- `jwt_jwks_path` - The JSON Web Key Set (JWKS) is a set of keys containing the public keys used to verify any JWT. These can be uploaded as entries in a single file. When configuring the flag in YugabyteDB Anywhere, click **Add JSON web key set (JWKS)** to upload the JWKS.
+
+The following illustration shows an example of setting the `ysql_hba_conf_csv` flag in YugabyteDB Anywhere:
+
+![Configuring ysql_hba_conf_csv flag for OIDC](/images/yp/security/oidc-jumpcloud-hbaconf.png)
+
+The following shows an example `ysql_hba_conf_csv` flag configuration for OIDC:
+
+```sh
+host all all 0.0.0.0/0 jwt_map=map1 jwt_audiences=""<client_id>"" jwt_issuers=""https://oauth.id.jumpcloud.com/.well-known/openid-configuration"" jwt_matching_claim_key=""preferred_username""
```
Comment (Arpit-yb) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:129]: Please remove `/.well-known/openid-configuration` from jwt_issuers.

**Review by Arpit-yb — CHANGES_REQUESTED** [review]

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/oidc-authentication.md [hunk 2]
```diff
@@ -41,6 +41,9 @@ To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super A
 **Learn more**
 
 - For information on configuring a YugabyteDB Anywhere universe to use OIDC-based authentication using Azure AD as the IdP, refer to [OIDC authentication with Azure AD](../../security/authentication/oidc-authentication-aad/).
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/oidc-authentication.md:43]:
```suggestion
- For information on configuring a YugabyteDB Anywhere universe to use OIDC-based authentication, refer to [OIDC authentication](../../security/authentication/oidc-authentication-aad/).
```

File: docs/content/preview/yugabyte-platform/administer-yugabyte-platform/oidc-authentication.md [hunk 3]
```diff
@@ -41,6 +41,9 @@ To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super A
...
+
+- For information on configuring a YugabyteDB Anywhere universe to use OIDC-based authentication using JumpCloud as the IdP, refer to [OIDC authentication with JumpCloud](../../security/authentication/oidc-authentication-jumpcloud/).
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/administer-yugabyte-platform/oidc-authentication.md:45]:
```suggestion
```

File: docs/content/preview/yugabyte-platform/security/authentication/_index.md [hunk 4]
```diff
@@ -31,8 +31,8 @@ YugabyteDB supports LDAP and OIDC for database authentication.
     icon="/images/section_icons/secure/authentication.png">}}
 
   {{<index/item
-    title="OIDC with Azure AD"
-    body="Authenticate database users using SSO via Azure AD."
+    title="OIDC authentication"
+    body="Authenticate database users using SSO via Azure AD or JumpCloud as the IdP."
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/_index.md:35]:
```suggestion
    body="Use an identity provider to authenticate database users."
```

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+    You will be prompted in a pop up to save the **Client ID** and **Client Secret**. Save these in a secure location, you will need to provide these credentials in YugabyteDB Anywhere.
+
+1. Configure Attributes and Identity Management as required.
+
+1. Integrate the user in JumpCloud.
+
+    - Navigate to **User Groups**, select the user groups you want to access YugabyteDB Aeon, and click **Save** when you are done.
+
+To configure JumpCloud federated authentication in YugabyteDB Aeon, you need the following application properties:
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:77]: Aeon? Delete this bit I think

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+After OIDC is set up, users can sign in to the YugabyteDB universe database using their JSON Web Token (JWT) as their password.
+
+Note that the yugabyte privileged user will continue to exist as a local database user even after OIDC-based authentication is enabled for a universe.
+
+**Learn more**
+
+- [Enable YugabyteDB Anywhere authentication via OIDC](../../../administer-yugabyte-platform/oidc-authentication/)
+- [YFTT: OIDC Authentication in YSQL](https://www.youtube.com/watch?v=KJ0XV6OnAnU&list=PL8Z3vt4qJTkLTIqB9eTLuqOdpzghX8H40&index=1)
+
+## Create an application in JumpCloud
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:40]: Need an intro similar to Set up OIDC with Azure AD on YugabyteDB Anywhere heading on Azure tab.
And then fix the heading levels below accordingly - this should be H3

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    - **Client ID** and **Client Secret** - enter the client ID and secret of the JumpCloud application you created.
+    - **Discovery URL** - enter `https://oauth.id.jumpcloud.com/.well-known/openid-configuration`.
+    - **Scope** - enter `openid email`.
+    - **Email attribute** - enter your registered email.
+
+1. Click **Save**.
+
+You are redirected to sign in to your IdP to test the connection. After the test connection is successful, OIDC authentication is enabled.
+
+### Configure a universe
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:101]: Are these section that follow identical to Azure?

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+- In YugabyteDB Anywhere, create YBA users.
+
+  Create a user in YugabyteDB Anywhere for each user who wishes to sign in to YBA to obtain their JWT.
+
+  To view their JWT, YBA users can sign in to YugabyteDB Anywhere, click the **User** icon at the top right, select **User Profile**, and click **Fetch OIDC Token**.
+
+  This is not required if you enabled the **Display JWT token on login** option in the YBA OIDC configuration, as any database user can copy the JWT from the YBA landing page without signing in to YBA.
+
+  For information on how to add YBA users, see [Create, modify, and delete users](../../../administer-yugabyte-platform/anywhere-rbac/#create-modify-and-delete-users).
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:192]: What about "Using your JWT"? doesn't that also apply?
Should these sections be done with a include?

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+#### Enable OIDC authentication
+
+To configure User authentication in YugabyteDB Anywhere, do the following:
+
+1. Navigate to **Admin > Access Management > User Authentication** and select **ODIC configuration**.
+1. Under **OIDC configuration**,  configure the following:
+
+    - **Client ID** and **Client Secret** - enter the client ID and secret of the JumpCloud application you created.
+    - **Discovery URL** - enter `https://oauth.id.jumpcloud.com/.well-known/openid-configuration`.
+    - **Scope** - enter `openid email`.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:94]: See also DOC-432

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md (hunk 7 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:101]: they are identical with minor modifications.

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md (hunk 9 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md]: addressed this too

**Review by Arpit-yb — APPROVED** [review]
LGTM!

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 10]
```diff
@@ -154,21 +167,15 @@ To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super A
...
-    ```
-
-1. Set the scope to `openid email profile`.
-
-1. Set the email attribute to a name for the property to be used in the mapping file, such as `preferred_username`.
+1. Navigate to **Admin > Access Management > User Authentication** and select **ODIC configuration**.
+1. Under **OIDC configuration**,  configure the following:
 
-1. Select the **Display JWT token on login** option to allow users to access their JWT from the YugabyteDB Anywhere sign in page. This allows a user to view and copy their JWT without signing in to YBA. (This option is only available if you enabled the `yb.security.oidc_feature_enhancements` configuration flag.)
+    - **Client ID** and **Client Secret** - enter the client ID and secret of the JumpCloud application you created.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:173]: this is Azure page, why Jumpcloud

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+    - Navigate to **User Groups**, select the user groups you want to access YugabyteDB Anywhere, and click **Save** when you are done.
+
+To configure JumpCloud federated authentication in YugabyteDB Anywhere, you need the following application properties:
+
+- **Client ID** and **Client Secret** of the application you created. These are the credentials you saved when you activated your application. The **Client ID** is also displayed on the **SSO** tab.
+
+For more information, refer to the [JumpCloud](https://jumpcloud.com/support/sso-with-oidc) documentation.
+
+### Configure YugabyteDB Anywhere
+
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:89]: Do you need to Enable OIDC enhancements for JumpCloud (like in Azure)
Azure page says "To allow users to access their JWT from the YugabyteDB sign in page, you must enable the OIDC feature via a configuration flag before you configure OIDC." so seems lile it?

Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:89]: To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super Admin. You need your JumpCloud application client ID and client secret.

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md [hunk 12]
```diff
@@ -154,21 +167,15 @@ To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super A
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:167]: Would put this "To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super Admin. You need your Azure application client ID, client secret, and tenant ID." under Configure YugabyteDB Anywhere

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md (hunk 10 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:173]: mistook it to be the jumpcloud page.. corrected it

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md (hunk 11 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:89]: Yes I believe so. @Arpit-yb had mentioned that the steps are very similar.

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+1. Navigate to **Admin > Advanced > Global Configuration**.
+
+1. Search on OIDC to display the configuration setting and set it to true.
+
+    ![Configuring yb.security.oidc_feature_enhancements flag for OIDC](/images/yp/security/oidc-azure-globalfeature.png)
+
+#### Enable OIDC authentication
+
+To configure YugabyteDB Anywhere for OIDC, you need to be signed in as a Super Admin. You need your JumpCloud application client ID and client secret.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md:104]: Would move to Configure YugabyteDB Anywhere cause you need superuser to set the global flag also I believe

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md (hunk 12 above)
Comment (ddhodge) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-aad.md:167]: Would still move this paragraph up to Configure YugabyteDB Anywhere cause I think you need superadmin to set the global flag as well

File: docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md (hunk 13 above)
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/security/authentication/oidc-authentication-jumpcloud.md]: aah yes .. I made the change to azure as well.

**Review by ddhodge — APPROVED** [review]

**Pull request merged**